### PR TITLE
♻️ (losses): Sub-parameters separated by . not _

### PR DESCRIPTION
**Pull request by vict0rsch**
Address #62

### CODE REVIEW

File: scripts/train_hypergrid.py
Comment (saleml) [inline scripts/train_hypergrid.py:1]: is this just:
1 - `if __name__ == "__main__":`
2- add `help` everywhere ?
If so, I'll handle the merge

Comment (vict0rsch) [reply scripts/train_hypergrid.py:1]: Yes and [`2d865f2` (#64)](https://github.com/saleml/torchgfn/pull/64/commits/2d865f2453bf48a6664dca11d4219643bec5bb7f)

File: .gitignore [hunk 1]
```diff
@@ -99,7 +99,7 @@ ipython_config.py
 #   This is especially recommended for binary packages to ensure reproducibility, and is more
 #   commonly ignored for libraries.
 #   https://python-poetry.org/docs/basic-usage/#commit-your-poetrylock-file-to-version-control
-#poetry.lock
+poetry.lock
```
Comment (josephdviviano) [inline .gitignore:102]: swanky!

File: scripts/train_hypergrid.py [hunk 2]
```diff
@@ -30,205 +30,299 @@
...
+    #       three (same, + logZ) estimators for TB.
+    if args.loss == "FM":
+        # We need a LogEdgeFlowEstimator
+        if args.tabular:
+            module = Tabular(n_states=env.n_states, output_dim=env.n_actions)
+        else:
+            module = NeuralNet(
                 input_dim=env.preprocessor.output_dim,
-                output_dim=env.n_actions - 1,
+                output_dim=env.n_actions,
```
Comment (josephdviviano) [inline scripts/train_hypergrid.py:178]: What's going on here -- why can we get away with removing the `-1`?

File: scripts/train_hypergrid.py [hunk 3]
```diff
@@ -30,205 +30,299 @@
...
+                for key, val in parametrization.parameters.items()
+                if "logZ" not in key
+            ],
+            "lr": args.lr,
         }
-    )
-
-optimizer = torch.optim.Adam(params)
+    ]
+    if "logZ.logZ" in parametrization.parameters:
```
Comment (josephdviviano) [inline scripts/train_hypergrid.py:285]: is this what the PR title is referring to?

**Review by josephdviviano — COMMENTED** [review]
Looks good - only issue is I'm worried we accidentally removed a `-1` that would be a breaking change - can you confirm please?

**Comment by vict0rsch** [comment]
Wait that must be a mistake, I don't even remember doing that  -1.

@saleml we _must_ keep it right?

**Comment by saleml** [comment]
Yeah we must keep it. I don't think you removed it. It's just Github comparer being silly. I'll handle it in the merge.

**Pull request merged**